internals/overlord/backend.go

```diff
@@ -25,9 +25,15 @@ type overlordStateBackend struct {
 	path           string
 	ensureBefore   func(d time.Duration)
 	requestRestart func(t restart.RestartType)
+
+	// If this is true, the backend will not actually save anything to file.
+	DryRun bool
 }
 
 func (osb *overlordStateBackend) Checkpoint(data []byte) error {
+	if osb.DryRun {
```

Comment:
Instead of checking a flag in `Checkpoint`, could we provide a different type for the state backend, a do-nothing implementation?